tests/unit/schema/test_schema_io.py

```diff
@@ -32,6 +33,7 @@ def test_json_serialization_with_embedded_dicts():
         [
             ColumnSchema(
                 "userid",
+                dtype=dtype.int32,
```

Comment:
These changes are due to the stricter validation of schemas for IO; we now need column schemas with dtypes specified. This used to work without, but that was somewhat accidental: the dtype `None` was translated by Numpy to `np.float64`, which then meant we had an incorrect dtype to write to the file, which was still enough for these tests to pass since they're not particular about the dtypes.